compiler/optimizer/RematTools.cpp

```diff
@@ -316,3 +323,25 @@ void RematTools::walkTreeTopsCalculatingRematFailureAlternatives(TR::Compilation
       start = start->getNextTreeTop();
       }
    }
+
+/*
+ * This method is used to determine if indirect loads are safe for the nodes between
+ * the start inclusive and the end exclusive. It scans for indirect stores and
+ * determines the loads to be safe if none are found. The result can be passed to
+ * gatherNodesToCheck to allow remat of stores dependent on indirect loads.
+ */
+bool RematTools::checkForSafeIndirectLoads(TR::Compilation *comp, TR::TreeTop *start,
```

Comment:
Don't we also need to check for volatile reads/writes which create happens before relationships in the memory model which could make the remat invalid as well.